bundle-workflow/python/lib/bundle.py

```diff
@@ -0,0 +1,49 @@
+#  SPDX-License-Identifier: Apache-2.0
+#
+#  The OpenSearch Contributors require contributions made to
+#  this file be licensed under the Apache-2.0 license or a
+#  compatible open source license.
+
+import os
+import tarfile
+import subprocess
+import shutil
+import tempfile
+from dataclasses import dataclass
+from lib.bundle_component import BundleComponent
+
+
+@dataclass
+class Bundle:
+    min_bundle: BundleComponent
+    dir: tempfile.TemporaryDirectory = None
+    path: str = None
+
+    def __post_init__(self):
+        self.dir = tempfile.TemporaryDirectory()
+        dir_name = self.dir.name
+        self.min_bundle.download(dir_name)
+        self.unpack(dir_name)
+        # OpenSearch & Dashboard tars will include only a single folder at the top level
+        self.path = next(iter([file.path for file in os.scandir(dir_name) if file.is_dir()]))
+
+    def install_plugin(self, plugin):
+        cli_path = os.path.join(self.path, 'bin/opensearch-plugin')
+        self.execute(f'{cli_path} install --batch file:{plugin.local_path}')
+
+    def execute(self, command):
+        print(f'Executing "{command}" in {self.path}')
+        subprocess.check_call(command, cwd = self.path, shell = True)
+
+    def unpack(self, dest):
+        tar = tarfile.open(self.min_bundle.local_path)
+        tar.extractall(dest)
+        tar.close()
+
+    def list_installed_plugins(self):
```

Comment:
Turn this into a property that returns a list of plugin paths, `print` outside of this class